### ISSUE TITLE
meet ErrSchemaSnapshotNotFound error in TiCDC 4.0.9

### ISSUE THREAD

**Issue by amyangfei** [pingcap/ticdc#1235]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

    changefeed meets error

2. What did you expect to see?

3. What did you see instead?

```
[2020/12/24 08:32:44.792 +00:00] [ERROR] [processor.go:1266] ["error on running processor"] [captureid=e42540a4-6bcf-4b8e-bc53-80bf02a6f71e] [captureaddr=tidb-cluster01-ticdc-5.tidb-cluster01-ticdc-peer.tidb-cluster01.svc:8301] [changefeedid=user-common-offline-5] [processorid=7deca114-d1da-4036-b20a-36b3dd75cb41] [error="[CDC:ErrSchemaSnapshotNotFound]can not found schema snapshot, ts: 421736913222762923"] [errorVerbose="[CDC:ErrSchemaSnapshotNotFound]can not found schema snapshot, ts: 421736913222762923
github.com/pingcap/errors.AddStack
        github.com/pingcap/errors@v0.11.5-0.20201029093017-5a7df2af2ac7/errors.go:174
github.com/pingcap/errors.(*Error).GenWithStackByArgs
        github.com/pingcap/errors@v0.11.5-0.20201029093017-5a7df2af2ac7/normalize.go:156
github.com/pingcap/ticdc/cdc/entry.(*SchemaStorage).getSnapshot
        github.com/pingcap/ticdc/cdc/entry/schema_storage.go:684
github.com/pingcap/ticdc/cdc/entry.(*SchemaStorage).GetSnapshot.func1
        github.com/pingcap/ticdc/cdc/entry/schema_storage.go:702
github.com/pingcap/ticdc/pkg/retry.RunWithInfiniteRetry.func1
        github.com/pingcap/ticdc/pkg/retry/retry.go:50
github.com/cenkalti/backoff.RetryNotify
        github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37
github.com/cenkalti/backoff.Retry
        github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24
github.com/pingcap/ticdc/pkg/retry.RunWithInfiniteRetry
        github.com/pingcap/ticdc/pkg/retry/retry.go:49
github.com/pingcap/ticdc/cdc/entry.(*SchemaStorage).GetSnapshot
        github.com/pingcap/ticdc/cdc/entry/schema_storage.go:695
github.com/pingcap/ticdc/cdc/entry.(*mounterImpl).unmarshalAndMountRowChanged
        github.com/pingcap/ticdc/cdc/entry/mounter.go:236
github.com/pingcap/ticdc/cdc/entry.(*mounterImpl).codecWorker
        github.com/pingcap/ticdc/cdc/entry/mounter.go:187
github.com/pingcap/ticdc/cdc/entry.(*mounterImpl).Run.func2
        github.com/pingcap/ticdc/cdc/entry/mounter.go:164
golang.org/x/sync/errgroup.(*Group).Go.func1
        golang.org/x/sync@v0.0.0-20200625203802-6e8e738ad208/errgroup/errgroup.go:57
runtime.goexit
        runtime/asm_amd64.s:1373"]
```

```
"id": "user-common-offline-5",
    "summary": {
      "state": "stopped",
      "tso": 421736913314513630,
      "checkpoint": "2020-12-24 08:30:50.034",
      "error": {
        "addr": "tidb-cluster01-ticdc-5.tidb-cluster01-ticdc-peer.tidb-cluster01.svc:8301",
        "code": "CDC:ErrProcessorUnknown",
        "message": "[CDC:ErrSchemaSnapshotNotFound]can not found schema snapshot, ts: 421736913222762923"
```

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        4.0.7
        ```

    - TiCDC version (execute `cdc version`):

        ```
        4.0.9
        ```

**Comment by liuzix** [comment]
I opened an ad hoc fix PR #1236 

The error might have been caused by the following order of events:

1. Owner calculates checkpoint ts and modifies local state. Owner changes the local copy of checkpoint ts from A to B.
2. Owner handles DDL jobs and dispatches a table that has been created at B.
3. Owner calls `rebalance` and creates the relevant task key in Etcd.
4. Since the target capture has just been created, and the processor has not been run yet. The capture sees the Etcd key, and starts the processor. Note that the checkpoint ts in Etcd is still A.
5. The processor is started with start-ts A, and the local checkpoint ts of the capture is A.
6. Before the DDL puller in the processor catches up with the checkpoint ts B, the processor exits abnormally.
7. Owner sees that the capture has been deleted, and adds the table to orphan table, meanwhile adjusting the start-ts of that table to the local checkpoint ts of the capture that has just exited, i.e. A.
8. Owner dispatches the table to another capture.
9. The other processor starts the table from A. Note that A < B.
10. If the other processor has been started from a ts greater than A, it will not have schemas at A, hence the error.

**Comment by amyangfei** [comment]
Is it possible to add an integration test to reproduce this bug?
Besides, how can we distinguish the reason from log quickly, do we need to add more logs for diagnostic
